app/models/solidus_subscriptions/user_generator.rb

```diff
@@ -19,10 +19,12 @@ def find_or_create(order)
     private
 
     def existing_user(order)
-      # I tried using `find_by` here but that has a strange conflict with
-      # using `stub_authorization` in specs. `stub_authorization` adds an
-      # `allow` on `Spree::User` that specifically targets `find_by`.
-      Spree.user_class.where(email: order.email).first
+      user_attrs = { email: order.email }.tap do |u_attrs|
+        if Spree.user_class.new.respond_to?(:team_id)
+          u_attrs[:team_id] = order.store.team_id
+        end
+      end
+      Spree.user_class.where(user_attrs).first
```

Comment:
I recommend `find_by` or at last `take` in place of `first` (which is what find_by does under the hood). Look at the difference in the SQL generated by these:
`User.where(email: 'f@example.com').first`
```
SELECT "users".* FROM "users" WHERE "users"."email" = 'f@example.com' ORDER BY "users"."id"` ASC LIMIT 1
```
`User.where(email: 'f@example.com').take`
```
SELECT "users".* FROM "users" WHERE "users"."email" = 'f@example.com' LIMIT 1
```

I might be venturing into "you don't know what you're talking about space here," but I think `find_by`/`take` is better here.

Plagarized from: https://stackoverflow.com/questions/18496421/take-vs-first-performance-in-ruby-on-rails

Reply:
Someone recently made a similar change to Solidus that resulted in *seconds* difference in load time for a page.

Reply:
Interesting! I always appreciate optimization suggestions like this.

Reply:
Change made.